crates/catalog/hms/src/schema.rs

```diff
@@ -0,0 +1,376 @@
+        value: String,
+    ) -> iceberg::Result<String> {
+        Ok(value)
+    }
+
+    fn before_struct_field(
+        &mut self,
+        _field: &iceberg::spec::NestedFieldRef,
+    ) -> iceberg::Result<()> {
+        self.depth += 1;
+        Ok(())
+    }
+
+    fn r#struct(
+        &mut self,
+        r#_struct: &iceberg::spec::StructType,
+        results: Vec<String>,
+    ) -> iceberg::Result<String> {
+        Ok(format!("struct<{}>", results.join(", ")))
+    }
+
+    fn after_struct_field(
+        &mut self,
+        _field: &iceberg::spec::NestedFieldRef,
+    ) -> iceberg::Result<()> {
+        self.depth -= 1;
+        Ok(())
+    }
+
+    fn field(
+        &mut self,
+        field: &iceberg::spec::NestedFieldRef,
+        value: String,
+    ) -> iceberg::Result<String> {
+        if self.is_inside_struct() {
+            return Ok(format!("{}:{}", field.name, value));
+        }
+
+        self.schema.push(FieldSchema {
+            name: Some(field.name.clone().into()),
+            r#type: Some(value.clone().into()),
+            comment: field.doc.clone().map(|doc| doc.into()),
+        });
+
+        Ok(value)
+    }
+
+    fn list(&mut self, _list: &iceberg::spec::ListType, value: String) -> iceberg::Result<String> {
+        Ok(format!("array<{}>", value))
+    }
+
+    fn map(
+        &mut self,
+        _map: &iceberg::spec::MapType,
+        key_value: String,
+        value: String,
+    ) -> iceberg::Result<String> {
+        Ok(format!("map<{},{}>", key_value, value))
+    }
+
+    fn primitive(&mut self, p: &iceberg::spec::PrimitiveType) -> iceberg::Result<String> {
+        let hive_type = match p {
+            PrimitiveType::Boolean => "boolean".to_string(),
+            PrimitiveType::Int => "int".to_string(),
+            PrimitiveType::Long => "bigint".to_string(),
+            PrimitiveType::Float => "float".to_string(),
+            PrimitiveType::Double => "double".to_string(),
+            PrimitiveType::Date => "date".to_string(),
+            PrimitiveType::Timestamp => "timestamp".to_string(),
+            PrimitiveType::Time | PrimitiveType::String | PrimitiveType::Uuid => {
+                "string".to_string()
+            }
+            PrimitiveType::Binary | PrimitiveType::Fixed(_) => "binary".to_string(),
+            PrimitiveType::Decimal { precision, scale } => {
+                format!("decimal({},{})", precision, scale)
+            }
+            _ => {
+                return Err(Error::new(
+                    ErrorKind::FeatureUnsupported,
+                    "Conversion from 'Timestamptz' is not supported",
+                ))
+            }
+        };
+
+        Ok(hive_type)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use iceberg::{
+        spec::{ListType, MapType, NestedField, Schema, StructType, Type},
+        Result,
+    };
+
+    use super::*;
+
+    #[test]
+    fn test_schema_with_nested_maps() -> Result<()> {
+        let schema = Schema::builder()
```

Comment:
nit: Iceberg schema could be converted from/to json, see [example](https://github.com/apache/iceberg-rust/blob/83cdff48f3ff4ab0bc6d2b39ce41a65ace4ee26b/crates/iceberg/src/spec/schema.rs#L856)

This may help to improve readability.

Reply:
sure, i can do that - thanks for the suggestions.